src/Silex/Provider/ValidatorServiceProvider.php

```diff
@@ -46,7 +48,21 @@ public function register(Container $app)
         };
 
         $app['validator.mapping.class_metadata_factory'] = function ($app) {
-            return new LazyLoadingMetadataFactory(new StaticMethodLoader());
+            $loader = null;
+
+            if ($app->offsetExists('validator.mapping.use_annotation') && $app['validator.mapping.use_annotation'] === true) {
+                $loader = new AnnotationLoader(new AnnotationReader());
+            } else {
+                $loader = new StaticMethodLoader();
+            }
+
+            $cache = null;
+
+            if ($app->offsetExists('validator.mapping.cache')) {
```

Comment:
1. You shouldn't use `$app->offsetExists('id')` to test if an entry exists 
but `isset($app['id'])`
2. You should always assign a default value to the parameters/services you 
are going to use. This way, you won't have to check if they have been 
defined whenever you need them:

```php
$app['validator.mapping.cache'] = null;

$app['validator.mapping.class_metadata_factory'] = function ($app) {
    // ...

    return new LazyLoadingMetadataFactory($loader, $app['validator.mapping.cache']);
};
```